src/renderers/WebGLRenderer.js

```diff
@@ -1430,11 +1421,24 @@ function WebGLRenderer( parameters ) {
 
 		}
 
-		const progUniforms = materialProperties.program.getUniforms();
+		const progUniforms = program.getUniforms();
 		const uniformsList = WebGLUniforms.seqWithValue( progUniforms.seq, uniforms );
```

Comment:
Maybe some more of the uniform-related data could be made a property of the program so it wouldn't need to be regenerated here on program change? But it's fine to leave this as a follow-up change.

Reply:
> But it's fine to leave this as a follow-up change.

I would prefer that. The PR is already complex and this change be changed later.